identity-iota/src/did/doc/iota_document.rs

```diff
@@ -223,28 +220,18 @@ impl IotaDocument {
       }
     }
 
-    let method = document
-      .authentication()
-      .head()
-      .and_then(|method| document.resolve_method_ref(method))
-      .ok_or(Error::MissingAuthenticationMethod)?;
-
-    Self::check_authentication(method)?;
-
-    // Ensure the authentication method DID matches the document DID
-    if method.id().did().authority() != did.authority() {
-      return Err(Error::InvalidDocumentAuthAuthority);
-    }
```

Comment:
This check was removed since it prevents a DID Document from being deactivated according to the IOTA DID method specification.

`IotaDocument::default_signing_key` is fallible now to account for the change.